### PR TITLE
Move UseExpressionBodyForLambda to Analyzers layer

### PR DESCRIPTION
**Pull request by Youssef1313**
@mavasani

### CODE REVIEW

**Comment by Youssef1313** [comment]
@mavasani Can you review please? Thanks!

File: src/Features/CSharp/Portable/UseExpressionBodyForLambda/UseExpressionBodyForLambdaCodeRefactoringProvider.cs [hunk 1]
```diff
@@ -2,27 +2,94 @@
...
+        {
+        }
+
+        public override async Task ComputeRefactoringsAsync(CodeRefactoringContext context)
+        {
+            var document = context.Document;
+            var cancellationToken = context.CancellationToken;
+
+            var optionProvider = await document.GetAnalyzerOptionsProviderAsync(cancellationToken).ConfigureAwait(false);
+            var optionValue = UseExpressionBodyForLambdaDiagnosticAnalyzer.GetCodeStyleOption(optionProvider);
```
Comment (mavasani) [inline src/Features/CSharp/Portable/UseExpressionBodyForLambda/UseExpressionBodyForLambdaCodeRefactoringProvider.cs:37]: Can we move all this static helper methods from `UseExpressionBodyForLambdaDiagnosticAnalyzer` that are used in the fixer and/or refactoring into a separate static helper type `UseExpressionBodyForLambdaHelpers`?

**Comment by mavasani** [comment]
@Youssef1313 Can you please also port the analyzer/code fix provider unit tests from https://github.com/dotnet/roslyn/tree/main/src/EditorFeatures/CSharpTest/UseExpressionBodyForLambda/UseExpressionBodyForLambdasAnalyzerTests.cs into the shared layer: https://github.com/dotnet/roslyn/tree/main/src/Analyzers/CSharp/Tests?

**Review by mavasani — COMMENTED** [review]
Overall LGTM. Just another refactoring suggestion and need to port tests to shared layer.

**Review by mavasani — APPROVED** [review]
Thank you @Youssef1313!

**Pull request merged**